.github/scripts/e2e_tests.sh

```diff
@@ -239,4 +239,62 @@ if [[ "$USER_EXISTS" != *"NoSuchEntity"* ]]; then
   exit 1
 fi
 
+# Step 13: Test deletion bucket with deletion policy set
+
+log_and_run echo "Applying Bucket Class with deletion policy and respective Bucket Claim..."
+log_and_run kubectl apply -f cosi-examples/bucketclass-deletion-policy.yaml
+log_and_run kubectl apply -f cosi-examples/bucketclaim-deletion-policy.yaml
+
+log_and_run echo "Listing all S3 buckets before deletion..."
+log_and_run aws s3 ls --endpoint-url "$S3_ENDPOINT"
+
+BUCKET_CLASS_NAME="delete-bucket-class"
+
+log_and_run echo "Verifying bucket creation with prefix '$BUCKET_CLASS_NAME'..."
+
+for ((i=1; i<=$ATTEMPTS; i++)); do
+  log_and_run aws --endpoint-url "$S3_ENDPOINT" s3 ls
+  BUCKET_TO_BE_DELETED=$(aws --endpoint-url "$S3_ENDPOINT" s3api list-buckets --query "Buckets[?starts_with(Name, '$BUCKET_CLASS_NAME')].Name" --output text)
+
+  if [ -n "$BUCKET_TO_BE_DELETED" ]; then
+    log_and_run echo "Bucket created with prefix '$BUCKET_CLASS_NAME': $BUCKET_TO_BE_DELETED"
+    break
+  else
+    log_and_run echo "Attempt $i: Bucket with prefix '$BUCKET_CLASS_NAME' not found. Retrying in $DELAY seconds..."
+    sleep $DELAY
+  fi
+done
+
+if [ -z "$BUCKET_TO_BE_DELETED" ]; then
+  log_and_run echo "Bucket with prefix '$BUCKET_CLASS_NAME' was not created."
+  exit 1
+fi
+
+log_and_run echo "Deleting Bucket Claim..."
+log_and_run kubectl delete -f cosi-examples/bucketclaim-deletion-policy.yaml
+
+# Check if the bucket with name $BUCKET_TO_BE_DELETED exists by doing a head bucket.
+# If bucket exists, retry with ATTEMPTS and DELAY. If bucket is not found, test success.
+
+log_and_run echo "Verifying bucket deletion with name '$BUCKET_TO_BE_DELETED'..."
+
+for ((i=1; i<=$ATTEMPTS; i++)); do
+  BUCKET_HEAD_RESULT=$(aws --endpoint-url "$S3_ENDPOINT" s3api head-bucket --bucket "$BUCKET_TO_BE_DELETED" 2>&1 || true)
+
+  if [[ "$BUCKET_HEAD_RESULT" == *"Not Found"* ]]; then
+    log_and_run echo "Bucket with name '$BUCKET_TO_BE_DELETED' not found. Bucket deletion successful."
+    break
+  else
+    log_and_run echo "Attempt $i: Bucket with name '$BUCKET_TO_BE_DELETED' still exists. Retrying in $DELAY seconds..."
+    sleep $DELAY
+  fi
+done
+
+if [[ "$BUCKET_HEAD_RESULT" != *"Not Found"* ]]; then
+  log_and_run echo "Bucket with name '$BUCKET_TO_BE_DELETED' was not deleted after $ATTEMPTS attempts."
+  exit 1
+fi
```

Comment:
The cli has a command to wait for a condition with retries: `aws s3api wait ...`

https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3api/wait/bucket-not-exists.html

And you can configure the retries with `max_attempts` and `retry_mode`: https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#retry-configuration

Reply:
hank you for the suggestion.

Unfortunately, the list-buckets API does not support this functionality. Since we only have the prefix Bucket_CLASS_NAME and not the exact bucket name, it cannot be used in this case.
- --retry and --delay: These options are not natively supported by the aws s3api list-buckets command. You would need to implement a custom retry logic, as demonstrated in the script above.
- aws s3api wait: This command is ideal for waiting on conditions such as bucket existence. However, it requires an exact bucket name and does not work with prefixes.

That being said, I have added support for retry logic in other cases, such as checking if a bucket is being deleted (when we know the exact bucket name) and verifying a user is being deleted (when we know the user’s name).

[Commit for bucket delete check](https://github.com/scality/cosi-driver/pull/61/commits/f6e5c90a25d5108136a22275dcb14756e23bc732)

[Commit for user delete check](https://github.com/scality/cosi-driver/pull/61/commits/13941bad39f39b785822394aea8b952d03ae2a12)

Reply:
Apologies for multiple comments, had some CI issues
FInally adding the passing commit link here for reference: https://github.com/scality/cosi-driver/pull/61/commits/7ff62b31272d4f8bc89da66c3fc1784179183771